### PR TITLE
Manual import fixes

### PR DESCRIPTION
**Pull request by hegerj**

### CODE REVIEW

**Comment by hegerj** [comment]
Work in progress, I will add solution to problem of automated nitrate cases having information in "manual" fields [1]. 
[1] https://issues.redhat.com/browse/RHELPLAN-39596?focusedCommentId=15104424&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-15104424

**Comment by psss** [comment]
It might make sense to put fixes which are not relevant to manual import (e.g. runtest.sh permission or rhts sourcing) into separate pull requests. In this way it's easier to do the review and those changes can be merged earlier.

**Comment by hegerj** [comment]
yeah, this PR is kinda messy, I'll try to fix it

**Comment by psss** [comment]
Squashed and rebased on the latest master.

File: tmt/convert.py [hunk 1]
```diff
@@ -95,16 +95,11 @@ def read_manual(plan_id, case_id, disabled):
...
-        md_content['action'] = html_to_markdown(testcase.action)
-        md_content['expected'] = html_to_markdown(testcase.effect)
-        md_content['cleanup'] = html_to_markdown(testcase.breakdown)
+        md_content = read_manual_data(testcase)
 
         # Test case metadata
         data = read_nitrate_case(testcase)
-        data['manual'] = True
+        data['manual'] = not testcase.automated
         data['test'] = 'test.md'
```
Comment (psss) [inline tmt/convert.py:103]: It will always be `True`, right? So the original statement was probably a bit more clear.

File: tmt/convert.py [hunk 2]
```diff
@@ -146,9 +151,9 @@ def write_markdown(path, content):
     try:
         with open(path, 'w', encoding='utf-8') as md_file:
             md_file.write(to_print)
+            echo(style(f"Test case successfully stored into '{path}'.", fg='magenta'))
     except IOError:
```
Comment (psss) [inline tmt/convert.py:155]: Please, break the long line.

File: tmt/convert.py [hunk 3]
```diff
@@ -366,9 +371,35 @@ def read_nitrate(beaker_task, common_data, disabled):
...
     individual_data = list()
+    md_content = dict()
     for testcase in testcases:
+        # Testcase data must be fetched due to
+        # https://github.com/psss/python-nitrate/issues/24
+        testcase._fetch()
         data = read_nitrate_case(testcase)
         individual_data.append(data)
+        # Check testcase for manual data
+        md_content_tmp = dict()
```
Comment (psss) [inline tmt/convert.py:382]: This line can be omitted?

File: tmt/convert.py [hunk 4]
```diff
@@ -366,9 +371,35 @@ def read_nitrate(beaker_task, common_data, disabled):
...
+        md_content_tmp = read_manual_data(testcase)
+        for key in md_content_tmp:
+            if md_content_tmp[key] != "":
+                md_content = md_content_tmp
```
Comment (psss) [inline tmt/convert.py:386]: What about using something like `if any(md_content_tmp.values())` here?

File: tmt/convert.py (hunk 1 above)
Comment (hegerj) [reply tmt/convert.py:103]: Probably, however if they aren't I wanted the variable to be consistent with the test selection (line 65).

File: tmt/convert.py (hunk 2 above)
Comment (hegerj) [reply tmt/convert.py:155]: ok

**Review by psss — CHANGES_REQUESTED** [review]
Seems to be working as expected. Proposed just a couple of minor enhancements.

File: tmt/convert.py (hunk 1 above)
Comment (psss) [reply tmt/convert.py:103]: I mean, in the case selection you only choose tests which have `test.automated == False`. So here it always must be `True`. Or am I missing anything?

File: tmt/convert.py (hunk 3 above)
Comment (hegerj) [reply tmt/convert.py:383]: yes, leftover from development

File: tmt/convert.py (hunk 1 above)
Comment (hegerj) [reply tmt/convert.py:103]: Yeah, you are right.

**Review by psss — APPROVED** [review]
Thanks for the adjustments. Rebased on the latest master and going to merge.

**Pull request merged**